cmd/gardenlet/app/app.go

```
@@ -460,6 +467,30 @@ func (g *garden) cleanupOrphanedExtensionsServiceAccounts(ctx context.Context, g
 	return flow.Parallel(taskFns...)(ctx)
 }
 
+// TODO(Kostov6): Remove this code after v1.89 has been released.
+func (g *garden) cleanupGRMSecretFinalizers(ctx context.Context, seedClient client.Client, log logr.Logger) error {
+	mrs := &resourcesv1alpha1.ManagedResourceList{}
+	if err := seedClient.List(ctx, mrs); err != nil {
+		log.Error(err, "Failed to list ManagedResources while cleaing up GRM finalizers")
+	}
+	secret := &corev1.Secret{}
+	for _, mr := range mrs.Items {
+		for _, ref := range mr.Spec.SecretRefs {
+			if err := g.mgr.GetClient().Get(ctx, client.ObjectKey{Namespace: mr.Namespace, Name: ref.Name}, secret); err != nil {
+				log.Error(err, "Failed to get secret while cleaing up GRM finalizers")
+			}
+			for _, finalizer := range secret.Finalizers {
+				if strings.HasPrefix(finalizer, predicate.FinalizerName) {
+					if err := controllerutils.RemoveFinalizers(ctx, seedClient, secret, finalizer); err != nil {
+						log.Error(err, "Failed to remove finalizer while cleaing up GRM finalizers")
```

return error instead of logging